radixdlt/src/main/java/com/radixdlt/consensus/Vertex.java

```diff
@@ -50,17 +53,55 @@
 	@DsonOutput(Output.ALL)
 	private final Atom atom;
 
+	private final transient Supplier<Hash> cachedHash;
+
 	Vertex() {
 		// Serializer only
 		this.qc = null;
 		this.round = null;
 		this.atom = null;
+		this.cachedHash = null;
 	}
 
-	public Vertex(QuorumCertificate qc, Round round, Atom atom) {
+	private Vertex(QuorumCertificate qc, Round round, Atom atom) {
 		this.qc = qc;
-		this.round = Objects.requireNonNull(round);
+		this.round = round;
 		this.atom = atom;
+		this.cachedHash = Suppliers.memoize(this::doGetHash);
+	}
+
+	public static Vertex createGenesis(Atom atom) {
+		return new Vertex(null, Round.of(0), atom);
+	}
+
+	public static Vertex createVertex(QuorumCertificate qc, Round round, Atom atom) {
+		Objects.requireNonNull(qc);
+
+		if (round.number() == 0) {
+			throw new IllegalArgumentException("Only genesis can have round 0.");
+		}
+
+		return new Vertex(qc, round, atom);
+	}
+
+	private Hash doGetHash() {
+		try {
+			return new Hash(Hash.hash256(Serialization.getDefault().toDson(this, Output.HASH)));
```

Comment:
Do we really want a static dependency on Serialization here?

Reply:
probably not, but using the pattern we've been using for all other classes is fine for now until we come up with better strategy.